src/index.js

```diff
@@ -272,7 +278,12 @@ export default class Carousel extends Component {
    * @param {String} direction - The direction to transition, should be 'right' or 'left'.
    */
   goToSlide(index, direction) {
-    const { beforeChange, transitionDuration, transition } = this.props;
+    const { beforeChange, transitionDuration, transition, onControlClick } = this.props;
+
+    if (onControlClick) {
+      onControlClick();
```

Comment:
should there be some sort of event passed along? Should it have the original click event data that could tell the consumer if they went forward/backwards/direct-to-index/etc.?

Reply:
I'm not sure if I follow, do you mean passing the event to the callback?

Reply:
possibly yeah, will consumers of this callback/event handler care about the original event? do they care _what_ was clicked on, etc. Beyond your immediate need, what would someone want from this callback?